temporal-sdk/src/main/java/io/temporal/internal/replay/ReplayWorkflowExecutor.java

```diff
@@ -89,6 +114,18 @@ public void eventLoop() {
   }
 
   private void completeWorkflow(@Nullable WorkflowExecutionException failure) {
+    Map<Long, SignalHandlerInfo> runningSignalHandlers =
+        workflow.getWorkflowContext().getRunningSignalHandlers();
+    if (!runningSignalHandlers.isEmpty()) {
+      log.warn(unfinishedSignalHandlesWarnMessage);
```

Comment:
In other SDKs we are showing the unfinished handlers. I think we should here too. May want to use a `LinkedHashMap` so we can preserve insertion order, or maybe even `LinkedHashSet` returning the info objects themselves to caller for later removal (will be identity equality on those info objects).

Reply:
Agree, as noted in the PR description this logic is not done yet. I am trying to find other java libraries the log a list of things as a MDC value.

Reply:
:+1: Let me know when to re-review

Reply:
Currently I have it logging the list as an ordered string, one open question I have is if it should be the default `ToString` or if I should format it as json?